core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -83,4 +85,25 @@ protected void deleteFiles(Set<String> pathsToDelete, String fileType) {
             (file, thrown) -> LOG.warn("Delete failed for {} file: {}", fileType, file, thrown))
         .run(deleteFunc::accept);
   }
+
+  protected Set<String> expiredStatisticsFilesLocations(
+      TableMetadata beforeExpiration, TableMetadata afterExpiration) {
+    Set<String> statisticsFilesLocationBeforeExpiry = allStatisticsFilesLocation(beforeExpiration);
+    Set<String> statisticsFilesLocationAfterExpiry = allStatisticsFilesLocation(afterExpiration);
+    statisticsFilesLocationBeforeExpiry.removeAll(statisticsFilesLocationAfterExpiry);
+    return statisticsFilesLocationBeforeExpiry;
+  }
+
+  private Set<String> allStatisticsFilesLocation(TableMetadata tableMetadata) {
+    Set<String> allStatisticsFilesLocation;
+    if (tableMetadata.statisticsFiles() != null) {
+      allStatisticsFilesLocation =
+          tableMetadata.statisticsFiles().stream()
+              .map(StatisticsFile::path)
+              .collect(Collectors.toSet());
+    } else {
+      allStatisticsFilesLocation = Sets.newHashSet();
+    }
+    return allStatisticsFilesLocation;
```

Comment:
Nit: I think we could just initialize `allStatisticsFile = Sets.newHashSet()`and remove the else block. Also newline after the block/before the return.

Reply:
done